stats/stats-server/src/update_service.rs

```diff
@@ -27,74 +31,98 @@ fn time_till_next_call(schedule: &Schedule) -> std::time::Duration {
+        let groups = self.charts.update_groups.values();
+        let init_update_tracker = InitialUpdateTracker::new(groups.len() as u64);
+        let mut group_update_jobs: FuturesUnordered<_> = groups
             .map(|group| {
                 let this = self.clone();
                 let group_entry = group.clone();
                 let default_schedule = default_schedule.clone();
-                let sema = semaphore.clone();
-                (
-                    async move {
-                        let _permit = sema.acquire().await.expect("failed to acquire permit");
+                let status_listener = self.status_listener.clone();
+                let initial_update_semaphore = initial_update_semaphore.clone();
+                let init_update_tracker = &init_update_tracker;
+                async move {
+                    if let Some(mut status_listener) = status_listener {
+                        let wait_result = status_listener
+                            .wait_until_status_at_least(
+                                group_entry.group.dependency_indexing_status_requirement(
+                                    &group_entry.enabled_members,
+                                ),
+                            )
+                            .await;
+                        if wait_result.is_err() {
+                            panic!("Indexing status listener channel closed");
+                        }
+                    }
+
+                    {
+                        let _init_update_permit = initial_update_semaphore
+                            .acquire()
+                            .await
+                            .expect("failed to acquire permit");
                         if let Some(force_full) = force_update_on_start {
                             this.clone().update(group_entry.clone(), force_full).await
                         };
-                    },
-                    self.spawn_group_updater(group.clone(), &default_schedule),
-                )
+                    }
+                    tracing::info!(
+                        update_group = group_entry.group.name(),
+                        "initial update for group is done"
+                    );
+                    init_update_tracker.mark_updated();
+                    init_update_tracker.report();
+                    this.run_recurrent_update(group_entry, &default_schedule)
+                        .await
+                }
             })
-            .collect::<(Vec<_>, Vec<_>)>();
-        futures::future::join_all(tasks).await;
-        tracing::info!("initial update is done");
+            .collect();
 
         let mut failed = 0;
-        while !updaters.is_empty() {
-            let (res, _, others) = futures::future::select_all(updaters).await;
-            updaters = others;
-            tracing::error!("updater stopped: {:?}", res);
-
+        while let Some(()) = group_update_jobs.next().await {
```

Comment:
maybe it was too late on the clock for me to review this but I spend like 10 minutes to understand why `while let Some(()) =` is considered as failed update, let's add comment like

```
// These futures should never complete because they run in infinite loop.
// If any completes, it means something went wrong.
```